webapp/channels/src/components/markdown/index.ts

```diff
@@ -34,7 +34,7 @@ function makeGetChannelNamesMap() {
         (state: GlobalState, props: Props) => props && props.channelNamesMap,
         (channelNamesMap, channelMentions) => {
             if (channelMentions) {
-                return Object.assign({}, channelMentions, channelNamesMap);
+                return Object.assign({}, channelNamesMap, channelMentions);
```

Comment:
Looking all the way back at https://github.com/mattermost/mattermost/pull/7324#pullrequestreview-60076025 where this was first added, the code is a bit different, but it looks like we wanted `channelNamesMap` to apply second so that the updated display name would be used if a channel is renamed. If we switch the order of these, we will fix this bug, but we'll break that behaviour.

Maybe we could make this code smarter about how it merges the objects instead of just using `Object.assign`? Each entry in `channelMentions` has the name of the team that that channel is on, so what if we build the returned object by:
1. If there's an entry in `channelNamesMap` for a channel and if the current team name matches the one in `channelMentions`, use the one in `channelNamesMap`
2. Otherwise, use the one from `channelMentions`

Hopefully that makes sense. Let me know if you have any questions

Reply:
got it will start working on it

Reply:
> If there's an entry in channelNamesMap for a channel and if the current team name matches the one in channelMentions, use the one in channelNamesMap

@hmhealey `channelNamesMap` doesn't contain current team name it only contains channel names

Reply:
Sorry, I mean `props.channelNamesMap` (the one that becomes `channelMentions`) has items which contain a `team_name` field. I don't know why the parameters of this function swap the name of that one since that makes this very confusing

Reply:
@hmhealey can you please review the PR now?